.github/workflows/add-or-validate-labels.yaml

```diff
@@ -0,0 +1,111 @@
+# Adds labels to pull requests based on the branch name. Labels are required by the "publish" workflow to determine
+name: 'Validate PR labels'
+
+on:
+  workflow_dispatch: # Lets you see what labels would be added, but doesn't actually add them because no PR triggered it
+  pull_request:
+    types:
+      - opened
+      - labeled
+      - unlabeled
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  add-labels: # Load the dependencies cache for all jobs
```

Comment:
Since this is consolidated now, I assume we want to run this not only on open anymore. This is where my lack of knowledge on GH workflows kick in. Would this cause any conflicts/issues if we manually update the labels to something that's different than the calculation? Like if the branch name was incorrectly named.

Reply:
No, this specific job that adds labels we do want to open only when the PR is opened. After that we can just listen for the label changes. And for the second part, if the user then changes the labels that were auto-added, the validation should still work as expected, and no conflicts should arise

Reply:
I see. I very conveniently missed line 20